moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -567,72 +571,149 @@ TEST_F(OneRobot, FK)
+  EXPECT_NEAR(fraction, (double)expected_absolute_revolute_jump_traj_len / (double)full_traj_len, 0.01);
+
+  // Indirect call using testJointSpaceJumps
+  full_traj_len = generateTestTraj(traj, robot_model, joint_model_group);
   fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(1.0, 1.0));
-  EXPECT_EQ(traj.size(), 3);  // traj should be cut
-  EXPECT_NEAR(fraction, 3. / 4., 0.01);
-
-  // ignore revolute joints
-  generateTestTraj(traj, robot_model, joint_model_group);
+  EXPECT_EQ(full_traj_len, expected_full_traj_len);                  // full traj should be 7 waypoints long
+  EXPECT_EQ(traj.size(), expected_absolute_revolute_jump_traj_len);  // traj should be cut
+  EXPECT_NEAR(fraction, (double)expected_absolute_revolute_jump_traj_len / (double)full_traj_len, 0.01);
+
+  // Ignore prismatic joints
+  full_traj_len = generateTestTraj(traj, robot_model, joint_model_group);
+  fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(1.0, 0.0));
+  EXPECT_EQ(full_traj_len, expected_full_traj_len);                  // full traj should be 7 waypoints long
+  EXPECT_EQ(traj.size(), expected_absolute_revolute_jump_traj_len);  // traj should be cut
+  EXPECT_NEAR(fraction, (double)expected_absolute_revolute_jump_traj_len / (double)expected_full_traj_len, 0.01);
+
+  // Ignore revolute joints
+  full_traj_len = generateTestTraj(traj, robot_model, joint_model_group);
   fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(0.0, 1.0));
-  EXPECT_EQ(traj.size(), 4);  // traj should not be cut
-  EXPECT_NEAR(fraction, 4. / 4., 0.01);
+  EXPECT_EQ(full_traj_len, expected_full_traj_len);                   // full traj should be 7 waypoints long
+  EXPECT_EQ(traj.size(), expected_absolute_prismatic_jump_traj_len);  // traj should be cut
+  EXPECT_NEAR(fraction, (double)expected_absolute_prismatic_jump_traj_len / (double)expected_full_traj_len, 0.01);
+
+  // Ignore all joints
+  full_traj_len = generateTestTraj(traj, robot_model, joint_model_group);
+  fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(0.0, 0.0));
+  EXPECT_EQ(full_traj_len, expected_full_traj_len);  // full traj should be 7 waypoints long
+  EXPECT_EQ(traj.size(), expected_full_traj_len);    // traj should be cut
+  EXPECT_NEAR(fraction, 1.0, 0.01);
 }
 
 TEST_F(OneRobot, testRelativeJointSpaceJump)
 {
-  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("base_from_base_to_tip");
+  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("base_from_base_to_e");
   std::vector<std::shared_ptr<robot_state::RobotState>> traj;
 
-  // direct call of absolute version: factor slightly smaller than 3 (1.01 > 2.99 * 1.01 / 3)
```

Comment:
Why did you changed the threshold factor back to 1.0? This is not borderline anymore!